### PR TITLE
kth order statistic divide and conquer algorithm

### PR DESCRIPTION
**Pull request by choyiny**
### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: divide_and_conquer/kth_order_statistic.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    # + pivot (kth element)
+    # + big (elements larger than k)
+    if len(small) == k - 1:
+        return pivot
+    # pivot is in elements bigger than k
+    elif len(small) < k - 1:
+        return kth_number(big, k - len(small) - 1)
+    # pivot is in elements smaller than k
+    else:
+        return kth_number(small, k)
```
Comment (xcodz-dot) [inline divide_and_conquer/kth_order_statistic.py:58]:
```suggestion
        return kth_number(small, k)

```
PEP 8 fix

Comment (choyiny) [reply divide_and_conquer/kth_order_statistic.py:58]: I can assure you that I have a newline at the end of file.

Comment (xcodz-dot) [reply divide_and_conquer/kth_order_statistic.py:58]: sorry, i did not know that

**Comment by choyiny** [comment]
Looking to get a `hacktoberfest-accepted` tag if possible. @poyea

File: divide_and_conquer/kth_order_statistic.py (hunk 1 above)
Comment (poyea) [inline divide_and_conquer/kth_order_statistic.py:58]:
```suggestion
        return kth_number(small, k)

if __name__ == "__main__":
    import doctest

    doctest.testmod()
```

**Comment by poyea** [comment]
@choyiny make sure you add the tag `hacktoberfest` in your fork.

**Review by poyea — APPROVED** [review]
Thank you for your pull request!🤩

**Pull request merged**